src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JSImportExportTest.cs

```diff
@@ -31,6 +32,19 @@ public async Task MultipleImportAsync()
             Assert.Same(instance1, instance2);
         }
 
+        [Fact]
+        public async Task CancelableImportAsync()
+        {
+            var cts = new CancellationTokenSource();
+            var exTask = Assert.ThrowsAsync<JSException>(async () => await JSHost.ImportAsync("JavaScriptTestHelper", "./JavaScriptTestHelper.mjs", cts.Token));
+            cts.Cancel();
```

Comment:
- Would it be useful add a short delay before the Cancel, to give a chance for `ImportAsync` get started?
- Also, would it be useful to have another test that tries to import from an external url?
- What about a test for importing a non-existent file - like `./non-existent.js`, and a url `https://www.foobar.com/non-existent.js`, and canceling for those?

Reply:
The file is already in the browser cache so the promise would be fulfilled immediately and we would not see any cancelation at all.

I created ticket for it https://github.com/dotnet/runtime/issues/80309

Reply:
BTW the same infrastructure is used to cancel our WS on wasm. 
There are tests there which handle delayed cancelation/abort.
`src\libraries\System.Net.WebSockets.Client\tests\CancelTest.cs`
`src\libraries\System.Net.WebSockets.Client\tests\AbortTest.cs`